interval.hh

```diff
@@ -615,8 +615,12 @@ public:
     }
     // Takes a vector of possibly overlapping intervals and returns a vector containing
     // a set of non-overlapping intervals covering the same values.
-    template<typename Comparator>
-    static std::vector<nonwrapping_interval> deoverlap(std::vector<nonwrapping_interval> intervals, Comparator&& cmp) {
+    template<typename Comparator, typename IntervalVec>
+    requires requires (IntervalVec vec) {
```

Comment:
It's not your fault of course, but this "requires requires" has to be the ugliest feature of this already ugly language ;-) https://stackoverflow.com/questions/54200988/why-do-we-require-requires-requires

Reply:
There's a workaround, extract the requirement into a named concept and use it here. This moves the two requires apart.